modules/outbound-command-event-handler/test/integration/application/process_bulk_quotes_callback.test.ts

```diff
@@ -0,0 +1,607 @@
+    expect(bulkStateThree.state).toBe(BulkTransactionInternalState.AGREEMENT_PROCESSING);
+
+    // Check that command handler published BulkQuotesRequested message
+    const hasBulkQuotesRequested = (domainEvents.find((e) => e.getName() === 'BulkQuotesRequestedDmEvt'));
+    expect(hasBulkQuotesRequested).toBeTruthy();
+
+    // Check that bulk batches have been created.
+    const bulkBatchIds = await bulkTransactionEntityRepo.getAllBulkBatchIds(bulkTransactionId);
+    expect(bulkBatchIds[0]).toBeDefined();
+
+    // Get randomly generated quoteId
+    const preBulkBatch = await bulkTransactionEntityRepo.getBulkBatch(bulkTransactionId, bulkBatchIds[0]);
+    const bulkQuoteId = randomUUID();
+
+    // Simulate the domain handler sending ProcessBulkQuotesCallback to command handler
+    const processBulkQuotesCallbackCommandEventData : IProcessBulkQuotesCallbackCmdEvtData = {
+      bulkId: bulkTransactionId,
+      content: {
+        batchId: bulkBatchIds[0],
+        bulkQuoteId: bulkQuoteId,
+        bulkQuotesResult: {
+          bulkQuoteId: bulkQuoteId,
+          currentState: 'COMPLETED',
+          individualQuoteResults: [
+            {
+              quoteId: preBulkBatch.bulkQuotesRequest.individualQuotes[0].quoteId,
+              transferAmount: {
+                currency: 'USD',
+                amount: '123.45',
+              },
+              ilpPacket: 'string',
+              condition: 'string'
+            },
+            {
+              quoteId: preBulkBatch.bulkQuotesRequest.individualQuotes[0].quoteId,
+              transferAmount: {
+                currency: 'USD',
+                amount: '456.78',
+              },
+              ilpPacket: 'string',
+              condition: 'string'
+            }
+          ]
+        }
+      },
+      timestamp: Date.now(),
+      headers: []
+    }
+    const processBulkQuotesCallbackCommandEventObj = new ProcessBulkQuotesCallbackCmdEvt(
+      processBulkQuotesCallbackCommandEventData
+    );
+    await producer.sendCommandEvent(processBulkQuotesCallbackCommandEventObj);
+    await new Promise(resolve => setTimeout(resolve, messageTimeout));
+
+    // Check that the global state of bulk to be AGREEMENT_PROCESSING
+    const bulkStateFour = await bulkTransactionEntityRepo.load(bulkTransactionId);
+    expect(bulkStateFour.state).toBe(BulkTransactionInternalState.AGREEMENT_COMPLETED);
+
+    // Check that command handler published BulkQuotesCallbackProcessed message
+    const hasBulkQuotesCallbackProcessed = (domainEvents.find((e) => e.getName() === 'BulkQuotesCallbackProcessedDmEvt'));
+    expect(hasBulkQuotesCallbackProcessed).toBeTruthy();
+  });
+
+
+  // Skipping this since there is no easy way to programmatically control the
+  // maxEntryConfigPerBatch config value atm.
+  test.skip("Given Inbound command event ProcessBulkQuotesCallback for success requests \
+    Then the logic should update the individual batch state to AGREEMENT_PROCESSING, \
```

Comment:
Same comments as above. This TC description and the corresponding assertions need to be updated.